torch/jit/_ir_utils.py

```diff
@@ -0,0 +1,20 @@
+import torch
+from typing import Union
+
+class _InsertPoint(object):
+    def __init__(self, insert_point_graph: torch._C.Graph, insert_point: Union[torch._C.Node, torch._C.Block]):
+        self.insert_point = insert_point
+        self.g = insert_point_graph
+        self.guard = None
+
+    def __enter__(self):
+        self.prev_insert_point = self.g.insertPoint()
+        self.g.setInsertPoint(self.insert_point)
+
+    def __exit__(self, *args):
+        self.g.setInsertPoint(self.prev_insert_point)
+
+def with_insert_point(self, insert_point: Union[torch._C.Node, torch._C.Block]):
```

Comment:
Is `with g.with_insert_point` redundant? maybe `with g.insert_point_guard` or something else ? what do people think

Reply:
I like `insert_point_guard` slightly better